schema/validator.go

```diff
@@ -28,6 +30,12 @@ import (
 // and implements validation against a JSON schema.
 type Validator string
 
+type validateDescendantsFunc func(r io.Reader) error
+
+var mapValidateDescendants = map[Validator]validateDescendantsFunc{
+	MediaTypeManifest: validateManifestDescendants,
```

Comment:
@wking 
At beginning I work this in image-tool project, but seems improper. 
In `schema` package, `validateManifestDescendants` acting as the `validator.Validate` child is what I think the nice way. 
Callback handler must impact on up-layer calling in image-tool project, to which I worry maintainers not fully agree.